etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveMessageTypeStructure.java

```diff
@@ -0,0 +1,38 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleExecutionException;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+
+/** Removes the Message Structure (MSH-9.3) from the Message Type (MSH-9). */
+public class RemoveMessageTypeStructure implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args)
+            throws RuleExecutionException {
+        try {
+            Bundle bundle = (Bundle) resource.getUnderlyingResource();
+            Coding coding = HapiHelper.getMessageTypeCoding(bundle);
+            String display = coding.getDisplay();
+            String delimiter = "^";
+            ArrayList<String> displayList =
+                    Arrays.stream(display.split("\\" + delimiter))
+                            .collect(Collectors.toCollection(ArrayList::new));
+            if (displayList.size() < 3) {
+                return;
+            }
+            displayList.remove(2);
```

Comment:
This is just a contextual question - it seems like the other transforms are operating on entire FHIR fields, but this one's changing a substring of the display value of a `coding`.  Does the original MSH-9 not fill out the other parts of the `coding` (like the `system` and `code`) during the transition to FHIR inside ReportStream? How does this `display` field end up mapping back to MSH-9 on the other end when it goes back through ReportStream? I'm more familiar with FHIR in a FHIR-only context where the `display` would be like the least important part of a `coding`

Reply:
Good question. RS uses `code` to determine the message type, which is mapped from the original HL7 MSH-9. But not all the fields are parsed and mapped into separate properties, only `code` is AFAIK. `display` is mapped back to `MSH-9` as is (I think)